inc/saitam.h

```diff
@@ -910,6 +910,47 @@ typedef enum _sai_tam_telemetry_type_t
 
 } sai_tam_telemetry_type_t;
 
+typedef enum _sai_tam_tel_type_mode_t
+{
+    /**
+     * @brief This TAM telemetry type supports to bound multiple counter types
+     */
+    SAI_TAM_TEL_TYPE_MODE_MIXED_TYPE,
+
+    /**
+     * @brief This TAM telemetry type supports to bound only one counter type
+     */
+    SAI_TAM_TEL_TYPE_MODE_SINGLE_TYPE,
+
+} sai_tam_tel_type_mode_t;
+
+/**
+ * @brief TAM telemetry type state of state machine
+ */
+typedef enum _sai_tam_tel_type_state_t
+{
+    /**
+     * @brief Telemetry type is stopped
+     */
+    SAI_TAM_TEL_TYPE_STATE_STOPPED,
+
+    /**
+     * @brief Telemetry type is started
+     */
+    SAI_TAM_TEL_TYPE_STATE_STARTED,
+
+    /**
+     * @brief Telemetry type config is committed
+     */
+    SAI_TAM_TEL_TYPE_STATE_CONFIG_GENERATING,
+
+    /**
+     * @brief Telemetry type config is ready
+     */
+    SAI_TAM_TEL_TYPE_STATE_CONFIG_READY,
+
+} sai_tam_tel_type_state_t;
+
```

Comment:
Nitpik, names of the attributes are not clear about the intent of the state machine.
I belive what you are trying to do is as follows
Default State:  READY
Once all the subscription is completed, orchagent want to indiate to SAI that it can now start the creation of templates
Next State: CREATE
SAI adapter will do out of sync notification of template creation complete and once orchagent reads the templates then must transition to state to that hw should start streaming now
Next State: START
If there is any change in NOS or streaming need to be stopped then HW wil be instructed to stop streaming.
Next State: STOP
Once SAI adapter cleans its own data stucture etc, NOS will transition the state to ready
Next State: READY

FSM will look like this
READY -> CREATE - > START -> STOP -> READY
There are no intermediate state transitions.